R/sim_linelist.R

```diff
@@ -20,34 +20,90 @@
 #' * `proportion`: a column with the proportion of the population that are in
 #' that age group. Proportions must sum to one.
 #'
-#' @param contact_distribution An `<epidist>` object or anonymous function for
-#' the contact distribution. This is any discrete density function that
-#' produces non-negative integers (including zero, \eqn{\mathbb{N}_0}) for the
-#' number of contacts per infection.
-#' @param infectious_period An `<epidist>` object or anonymous function for
-#' the infectious period. This defines the duration from becoming infectious
-#' to no longer infectious. In the simulation, individuals are assumed to
+#' @param contact_distribution A `function` or an `<epidist>` object to generate
+#' the number of contacts per infection.
+#'
+#' The function can be defined or anonymous. The function must return a vector
+#' of `numerics`. The vector is the probability mass of sampling a certain
+#' number of contacts, i.e. the first element in the vector is the weight of
+#' sampling zero contacts, the second element is the weight of sampling a single
```

Comment:
I don't really understand this - if it's that why doesn't the user just pass a vector? From the default it seems this should instead be a function returning the probability mass of the argument?

Reply:
I guess the user could pass a vector. I thought that it would be easier for the user to pass a function as it would be familiar to parameterise in terms of Poisson, negative binomial, etc (and familiar given similarity to an offspring distribution). 

> From the default it seems this should instead be a function returning the probability mass of the argument?

I agree with this interpretation, does this contradict the documentation as currently stated?

Reply:
To me, yes. As stated it says

> the first element in the vector is the weight of sampling zero contacts

but the example suggests `function(x) stats::dpois(x = x, lambda = 2)` which is a function which if called e.g. with argument `3` gives a vector of length 1 with the probability of sampling 3 contacts as first element.

Reply:
The `contact_distribution` is always supplied `0:1e4 + 1` as it's argument:

https://github.com/epiverse-trace/simulist/blob/840b2723ee221282efa3e1928a82495cb402ba6d/R/sim_network_bp.R#L58

so it gives a vector of length 10,001, with the probability of sampling 0 to 10,000 contacts per case.

Would it help to rephrase to: _the first element in the vector returned from the function is the weight of sampling zero contacts ..._ ?

Reply:
No, I think to be consistent with the example you'd need to write something like "The function must take a single argument in the form of a an integer vector with elements representing numbers of contacts, and return a numeric vector where each element corresponds to the probability of observing the number of contacts in the vector passed to the function". (or some better rewrite)

The function passed in the example is not a function that always returns a vector where the first element corresponds to zero contacts etc., so it wouldn't be permissible according to the current description.

Reply:
Thanks for the suggested text. Resolved in 01441c5.